src/scalar/base64.h

```diff
@@ -9,21 +9,27 @@ namespace scalar {
 namespace {
 namespace base64 {
 
-// Returns true upon success. The destination buffer must be large enough and is
-// incremented by the number of bytes written and src is incremented by the number of bytes read.
+// Returns true upon success. The destination buffer must be large enough.
 // This functions assumes that the padding (=) has been removed.
-result base64_tail_decode(char *dst, const char *src, size_t length) {
-  const char *srcend = src + length;
-  const char *srcinit = src;
+template <class char_type>
+result base64_tail_decode(char *dst, const char_type *src, size_t length, base64_options options) {
+  const uint8_t *to_base64 = (options & base64_url) ? tables::base64::to_base64_url_value : tables::base64::to_base64_value;
+  const uint32_t *d0 = (options & base64_url) ? tables::base64::base64_url::d0 : tables::base64::base64_default::d0;
+  const uint32_t *d1 = (options & base64_url) ? tables::base64::base64_url::d1 : tables::base64::base64_default::d1;
+  const uint32_t *d2 = (options & base64_url) ? tables::base64::base64_url::d2 : tables::base64::base64_default::d2;
+  const uint32_t *d3 = (options & base64_url) ? tables::base64::base64_url::d3 : tables::base64::base64_default::d3;
```

Comment:
The four branch can be reduced into 1

Reply:
I expect that the compiler will do it for us...
https://godbolt.org/z/sc4ErzPKM
But you are correct that it looks inefficient at a glance.

I will add a remark that it is my expectation. That is, the reason to refactor this code would be for style or elegance, but I expect that the compiler is smart enough to optimize this away reasonably. It just looks expensive, I expect.